functionary/ui/forms/scheduled_task.py

```diff
@@ -52,11 +52,9 @@ class Meta:
             "parameters",
         ]
 
-    def __init__(self, *args, **kwargs):
-        environment = kwargs.pop("environment", None)
-        is_create = kwargs.pop("is_create", False)
+    def __init__(self, environment: Environment = None, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self._setup_field_choices(is_create)
+        self._setup_field_choices(kwargs.get("instance"))
```

Comment:
You're passing in an instance (or None) to `is_update` which is listed as a bool.  Probably just pass in `kwargs.get("instance") is not None`, and then everything lines up more cleanly.

Reply:
Fixed